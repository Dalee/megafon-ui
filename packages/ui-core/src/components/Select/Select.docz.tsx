import * as React from 'react';
import { ISelectItem } from './Select';

export const items = [
    {
        value: 1,
        title: 'Авиамоторная',
    },
    {
        value: 2,
        title: 'Академическая',
    },
    {
        value: 3,
        title: 'Бабушкинская',
    },
    {
        value: 4,
        title: 'Багратионовская',
    },
    {
        value: 5,
        title: 'ВДНХ',
    },
    {
        value: 6,
        title: 'Варшавская',
    },
    {
        value: 7,
        title: 'Деловой центр',
    },
    {
        value: 8,
        title: 'Дмитровская',
    },
    {
        value: 9,
        title: 'Калужская',
    },
    {
        value: 10,
        title: 'Кантемировская',
    },
];

export const flexStyle = {
    display: 'flex',
    flexWrap: 'wrap',
    marginLeft: '-15px',
};

const selectWrapperStyle = {
    height: '200px',
    width: '50%',
    paddingLeft: '15px',
    boxSizing: 'border-box' as 'border-box',
    minWidth: '300px',
};

type CurrentValue = number | undefined;

interface ISelectChildrenProps {
    currentValue: number | undefined;
    onSelect: (_e: React.SyntheticEvent<EventTarget>, data: ISelectItem) => void;
}

interface ISelectWrapperProps {
    children: (data: ISelectChildrenProps) => JSX.Element;
}

export const DemoSelectWrapper: React.FC<ISelectWrapperProps> = (props) => {
    const [currentValue, setCurrentValue] = React.useState<CurrentValue>(undefined);

    const handleSelect = (_e: React.SyntheticEvent<EventTarget>, data: ISelectItem) => {
        setCurrentValue(Number(data.value));
    };

    return (
        <div style={selectWrapperStyle}>
            {props.children({
                onSelect: handleSelect,
                currentValue: currentValue,
            })}
        </div>
    );
};
