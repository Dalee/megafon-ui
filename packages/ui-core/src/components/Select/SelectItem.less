@import '~styles/base.less';
@b: e('.mfui-select-item');

@{b} {
    .fontFamily();
    padding: 8px 32px;
    cursor: pointer;
    background-color: @clearWhite;

    &_active {
        background-color: @spbSky0;
    }

    &__title {
        display: flex;
        align-items: center;
        overflow: hidden;
        white-space: nowrap;
        flex-grow: 1;
        position: relative;

        &:after {
            content: '';
            position: absolute;
            top: 0;
            right: 0;
            bottom: 0;
            width: 40px;
            background: linear-gradient(to right, rgba(255, 255, 255, 0), @clearWhite);
        }
    }

    &_active &__title {
        &:after {
            background: linear-gradient(to right, rgba(255, 255, 255, 0), @spbSky0);
        }
    }
}
