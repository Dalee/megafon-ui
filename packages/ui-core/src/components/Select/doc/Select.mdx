---
name: Select
menu: Компоненты
route: components/select
---

import Example from './Select.example.mdx';
import Props from './Select.props.mdx';

# Select

<ImportExample name="Select" />

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
