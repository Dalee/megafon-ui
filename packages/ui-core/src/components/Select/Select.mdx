---
name: Select
menu: Components
route: components/select
---

import { Playground, Props } from 'docz';
import Select from './Select';
import Checked from 'icons/System/24/Checked_24.svg';
import { SelectWrapper, items, flexStyle } from './Select.docz';
import Metro from 'icons/System/24/Metro_24.svg';

# Select

## Properties

<Props of={Select} />

```javascript
const items = [
    {
        value: 1,
        view: 'Ватнайокюдль',
        title: 'Ватнайокюдль',
    },
    {
        value: 2,
        view: 'Хваннадальсхнукюр',
        title: 'Хваннадальсхнукюр',
    },
    {
        value: 3,
        view: 'Киркьюбайрклёйстюр',
        title: 'Киркьюбайклёйстюр',
    },
];
```

```javascript
const SelectWrapper = props => {
    const [currentValue, setCurrentValue] = React.useStat(undefined);

    const handleSelect = (_e, data) => {
        setCurrentValue(data.value);
    };

    return (
        <div style={selectWrapperStyle}>
            {props.children({
                onSelect: handleSelect,
                currentValue: currentValue,
            })}
        </div>
    );
};
```

## Select options

<Playground>
    <div style={flexStyle}>
        <SelectWrapper>
            {({ onSelect, currentValue }) => (
                <Select
                    label="Classic (default) type"
                    onSelect={onSelect}
                    placeholder="Choose your step"
                    items={items.map(item => ({
                        title: item.title,
                        value: item.value,
                    }))}
                    currentValue={currentValue}
                />
            )}
        </SelectWrapper>
        <SelectWrapper>
            {({ onSelect, currentValue }) => (
                <Select
                    label="Combobox type"
                    type="combobox"
                    onSelect={onSelect}
                    placeholder="Choose or type your step"
                    items={items.map(item => ({
                        title: item.title,
                        value: item.value,
                    }))}
                    currentValue={currentValue}
                />
            )}
        </SelectWrapper>
        <SelectWrapper>
            {({ onSelect, canOpen, currentValue }) => (
                <Select
                    label="Items with icon"
                    placeholder="Choose your step"
                    items={items.map(item => ({
                        title: item.title,
                        value: item.value,
                        view: (
                            <>
                                <Metro style={{ width: '24px', height: '24px', marginRight: '16px' }} />
                                {item.view}
                            </>
                        ),
                    }))}
                    onSelect={onSelect}
                    currentValue={currentValue}
                />
            )}
        </SelectWrapper>
        <SelectWrapper>
            {({ onSelect, canOpen, currentValue }) => (
                <Select
                    label="Zero items"
                    placeholder="Choose your step"
                    items={[]}
                    notFoundText="Nothing found"
                />
            )}
        </SelectWrapper>
        <SelectWrapper>
            {({ onSelect, currentValue }) => (
                <Select
                    label="Valid state"
                    verification="valid"
                    onSelect={onSelect}
                    placeholder="Choose your step"
                    items={items}
                    currentValue={currentValue}
                />
            )}
        </SelectWrapper>
        <SelectWrapper>
            {({ onSelect, currentValue }) => (
                <Select
                    label="Error state"
                    required
                    verification="error"
                    noticeText="Please, make a choice"
                    onSelect={onSelect}
                    placeholder="Choose your step"
                    items={items}
                    currentValue={currentValue}
                />
            )}
        </SelectWrapper>
        <SelectWrapper>
            {({ onSelect, currentValue }) => (
                <Select
                    label="Disabled state"
                    isDisabled
                    onSelect={onSelect}
                    placeholder="You have no choice"
                    items={items}
                    currentValue={currentValue}
                />
            )}
        </SelectWrapper>
    </div>
</Playground>
