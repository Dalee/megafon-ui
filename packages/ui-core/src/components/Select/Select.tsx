import * as React from 'react';
import * as PropTypes from 'prop-types';
import debounce from 'lodash.debounce';
import './Select.less';
import cnCreate from 'utils/cnCreate';
import detectTouch from 'utils/detectTouch';
import InputLabel from 'components/InputLabel/InputLabel';
import Paragraph from 'components/Paragraph/Paragraph';
import filterDataAttrs, { IDataAttributes } from './../../utils/dataAttrs';

export const Verification = {
    VALID: 'valid',
    ERROR: 'error',
} as const;

type VerificationType = typeof Verification[keyof typeof Verification];

export const SelectTypes = {
    CLASSIC: 'classic',
    COMBOBOX: 'combobox',
} as const;

type SelectTypesType = typeof SelectTypes[keyof typeof SelectTypes];

export type SelectItemValueType = number | string;

export interface ISelectItem {
    title: string;
    value: SelectItemValueType;
    view?: JSX.Element[] | Element[] | JSX.Element | string | Element;
}

interface ISelectClasses {
    control?: string;
    root?: string;
}

export interface ISelectProps extends IDataAttributes {
    /** Select type */
    type?: SelectTypesType;
    /** Field title */
    label?: string;
    /** Html id attribute for label */
    labelId?: string;
    /** Current selected item */
    currentValue?: SelectItemValueType;
    /** Verification */
    verification?: VerificationType;
    /** Notice text */
    noticeText?: string;
    /** isDisabled field */
    isDisabled?: boolean;
    /** Makes the field required. */
    required?: boolean;
    /** Placeholder */
    placeholder?: string;
    /** Text in the absence of search results */
    notFoundText?: string;
    /** Array of objects to be used for options rendering */
    items: ISelectItem[];
    /** Custom classname */
    className?: string;
    /** Object for the custom class */
    classes?: ISelectClasses;
    /** Select item handler */
    onSelect?: (e: React.MouseEvent<HTMLDivElement>, dataItem: ISelectItem) => void;
}

interface ISelectState {
    isOpened: boolean;
    activeIndex: number;
    filteredItems: ISelectItem[];
    comparableInputValue: string;
    inputValue: string;
}

const cn = cnCreate('mfui-beta-select');
class Select extends React.Component<ISelectProps, ISelectState> {
    static propTypes = {
        type: PropTypes.oneOf(Object.values(SelectTypes)),
        label: PropTypes.string,
        id: PropTypes.string,
        currentValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        verification: PropTypes.oneOf(Object.values(Verification)),
        noticeText: PropTypes.string,
        isDisabled: PropTypes.bool,
        required: PropTypes.bool,
        placeholder: PropTypes.string,
        notFoundText: PropTypes.string,
        className: PropTypes.string,
        classes: PropTypes.shape({
            control: PropTypes.string,
            root: PropTypes.string,
        }),
        items: PropTypes.arrayOf(
            PropTypes.shape({
                view: PropTypes.oneOfType([
                    PropTypes.string,
                    PropTypes.number,
                    PropTypes.element,
                ]),
                title: PropTypes.node,
                value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
            })
        ),
        onSelect: PropTypes.func,
        dataAttrs: PropTypes.objectOf(PropTypes.string.isRequired),
    };

    static defaultProps: Partial<ISelectProps> = {
        type: SelectTypes.CLASSIC,
        notFoundText: 'Ничего не нашлось',
        items: [],
    };

    itemWrapperNode: HTMLDivElement;
    itemsNodeList: HTMLDivElement[];
    selectNode: HTMLDivElement;

    isTouch: boolean = detectTouch();

    debouncedComboboxChange = debounce((filterValue: string) => {
        const { items } = this.props;

        const filteredItems = items.filter(({ title }) => {
            if (filterValue.length <= title.length) {
                return RegExp(filterValue, 'ig').test(title);
            }

            return false;
        });

        this.setState({ filteredItems, comparableInputValue: filterValue, isOpened: true });
    }, 250);

    constructor(props: ISelectProps) {
        super(props);

        this.state = {
            isOpened: false,
            activeIndex: 0,
            filteredItems: props.items,
            comparableInputValue: '',
            inputValue: '',
        };
        this.itemsNodeList = [];
    }

    componentDidUpdate() {
        const { isOpened } = this.state;

        if (isOpened) {
            document.addEventListener('click', this.handleClickOutside);

            return;
        }

        document.removeEventListener('click', this.handleClickOutside);
    }

    componentWillUnmount() {
        document.removeEventListener('click', this.handleClickOutside);
    }

    handleClickOutside = (e: MouseEvent): void => {
        const { isOpened } = this.state;

        if (e.target instanceof Node && this.selectNode.contains(e.target) || !isOpened) {
            return;
        }

        this.setState({ isOpened: false });
    }

    handleOpenDropdown = (): void => {
        this.setState((state) => ({ isOpened: !state.isOpened }));
    }

    handleClickItem = (itemValue: SelectItemValueType) => (e: React.MouseEvent<HTMLDivElement>): void => {
        const { onSelect, items } = this.props;
        const item = items.find(elem => elem.value === itemValue);

        if (!item) {
            return;
        }

        const { title } = item;

        this.setState({isOpened: false, inputValue: title });

        onSelect && onSelect(e, item);
    }

    handleHoverItem = (index: number) => (e: React.MouseEvent<HTMLDivElement>): void => {
        e.preventDefault();

        this.setState({ activeIndex: index });
    }

    handleClickCombobox = (e: React.FormEvent<EventTarget>): void => {
        const { isOpened, filteredItems } = this.state;

        e.stopPropagation();

        if (!(e.target instanceof HTMLInputElement)) {
            return;
        }

        if (!isOpened && filteredItems) {
            e.target.select();
        }

        this.setState((state) => ({ isOpened: !state.isOpened }));
    }

    handleChangeCombobox = (e: React.ChangeEvent<HTMLInputElement>): void => {
        const filterValue = e.target.value;

        this.setState({ inputValue: filterValue });

        this.debouncedComboboxChange(filterValue);
    }

    handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>): boolean => {
        const { activeIndex, isOpened } = this.state;
        const { items } = this.props;

        if (items.length === 0) {
            return true;
        }

        if (e.key === 'ArrowDown' && activeIndex < items.length - 1) {
            this.setState({ activeIndex: activeIndex + 1 }, () => {
                this.scrollList(this.state.activeIndex);
            });

            e.preventDefault();

            return false;
        }
        if (e.key === 'ArrowUp' && activeIndex > 0) {
            this.setState((prevState) => ({ activeIndex: prevState.activeIndex - 1 }), () => {
                this.scrollList(this.state.activeIndex);
            });

            e.preventDefault();

            return false;
        }
        if (e.key === 'Enter' && isOpened) {
            this.itemsNodeList[activeIndex].click();

            return false;
        }
        if (e.key === 'Enter' && !isOpened) {
            this.setState({ isOpened: true });

            return false;
        }
        if (e.key === 'Tab') {
            this.setState({ isOpened: false });

            return false;
        }

        return true;
    }

    scrollList(activeIndex: number): void {
        if (!this.itemsNodeList) {
            return;
        }

        const wrapper = this.itemWrapperNode;
        const wrapperScroll = wrapper.scrollTop;
        const wrapperHeight = wrapper.offsetHeight;

        const item = this.itemsNodeList[activeIndex];

        if (!item) {
            return;
        }

        const itemOffset = item.offsetTop;
        const itemHeight = item.offsetHeight;

        if (itemOffset + itemHeight > wrapperScroll + wrapperHeight) {
            wrapper.scrollTop = itemOffset + itemHeight - wrapperHeight;
        }

        if (itemOffset < wrapperScroll) {
            wrapper.scrollTop = itemOffset;
        }
    }

    highlightString = (title, view) => {
        const { type } = this.props;
        const { comparableInputValue } = this.state;

        if (type === SelectTypes.CLASSIC) {
            return view || title;
        }
        if (type === SelectTypes.COMBOBOX && view) {
            return view;
        }

        const stringFragments = title.split(RegExp(`(${comparableInputValue})`, 'ig'));

        return (
            <Paragraph hasMargin={false}>
                {stringFragments.map((fragment, i) => (
                    <React.Fragment key={i}>
                        {(fragment.toLowerCase() === comparableInputValue.toLowerCase())
                            ? <span className={cn('highlighted-fragment')}>{fragment}</span>
                            : fragment
                        }
                    </React.Fragment>
                ))}
            </Paragraph>
        );
    }

    getItemWrapper = node => this.itemWrapperNode = node;
    getSelectNode = node => this.selectNode = node;
    getNodeList = node => this.itemsNodeList.push(node);

    renderTitle() {
        const { placeholder, items, currentValue } = this.props;
        const item = items.find(elem => elem.value === currentValue);
        let inputTitle: string | undefined = placeholder;

        if (item && item.title) {
            inputTitle = item.title;
        }

        return (
            <div
                className={cn('title', {
                    placeholder: !!placeholder && !currentValue,
                })}
                tabIndex={0}
                onClick={this.handleOpenDropdown}
            >
                <div className={cn('title-inner')}>
                    {inputTitle}
                </div>
            </div>
        );
    }

    renderCombobox() {
        const { placeholder } = this.props;
        const { inputValue } = this.state;

        return (
            <input
                className={cn('combobox')}
                onClick={this.handleClickCombobox}
                onChange={this.handleChangeCombobox}
                type="text"
                value={inputValue}
                placeholder={placeholder}
            />
        );
    }

    renderChildren() {
        const { type, items, notFoundText } = this.props;
        const { filteredItems, activeIndex } = this.state;
        const currentItems = type === SelectTypes.COMBOBOX ? filteredItems : items;

        return (
            <div className={cn('list')}>
                <div className={cn('list-inner')} ref={this.getItemWrapper}>
                    {currentItems.map(({ title, value, view }, i) => (
                        <div
                            className={cn('list-item', {
                                active: activeIndex === i,
                            })}
                            key={`${i}_${value}`}
                            onClick={this.handleClickItem(value)}
                            onMouseEnter={this.handleHoverItem(i)}
                            ref={this.getNodeList}
                        >
                            <div className={cn('item-title')}>
                                {this.highlightString(title, view)}
                            </div>
                        </div>
                    ))}
                    {!currentItems.length && <div className={cn('not-found')}>{notFoundText}</div>}
                </div>
            </div>
        );
    }

    render() {
        const {
            type,
            isDisabled,
            verification,
            noticeText,
            label,
            labelId,
            required,
            className = '',
            classes = {},
            dataAttrs,
        } = this.props;
        const { isOpened } = this.state;

        return (
            <div
                {...filterDataAttrs(dataAttrs)}
                className={cn({
                    open: isOpened,
                    disabled: isDisabled,
                    'no-touch': !this.isTouch,
                    valid: verification === Verification.VALID,
                    error: verification === Verification.ERROR,
                }, [className, classes.root])}
                ref={this.getSelectNode}
            >
                <div className={cn('inner')}>
                    {label && (
                        <InputLabel htmlFor={labelId}>
                            {label}
                            {required && <span className={cn('require-mark')}>*</span>}
                        </InputLabel>
                    )}
                    <div
                        className={cn('control', classes.control)}
                        onKeyDown={this.handleKeyDown}
                    >
                        {(type === SelectTypes.COMBOBOX) && this.renderCombobox()}
                        {(type === SelectTypes.CLASSIC) && this.renderTitle()}
                        <div className={cn('arrow-wrap')} tabIndex={1} onClick={this.handleOpenDropdown}>
                            <span className={cn('arrow')} />
                        </div>
                    </div>
                    {this.renderChildren()}
                </div>
                {noticeText &&
                    <div className={cn('text', {
                        error: verification === Verification.ERROR,
                        success: verification === Verification.VALID,
                    })}>
                        {noticeText}
                    </div>
                }
            </div>
        );
    }
}

export default Select;
