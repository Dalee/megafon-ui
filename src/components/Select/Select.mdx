---
name: Select
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Select from './Select';

export const items = [{
    id: '1',
    value: 'eat',
    title: 'Eat',
}, {
    id: '2',
    value: 'sleep',
    title: 'Sleep',
}, {
    id: '3',
    value: 'traing',
    title: 'Traing',
}];

export const Wrapper = ({ children }) => {
    return (
        <div style={{ height: '200px' }}>
            {children}
        </div>
    );
}

# Select

## Properties
<PropsTable of={Select} />

``` javascript
const items = [{
    id: 1,
    value: 'eat',
    title: 'Eat',
}, {
    id: 2,
    value: 'sleep',
    title: 'Sleep',
}, {
    id: 3,
    value: 'traing',
    title: 'Traing',
}];
```

export class SelectWrapper extends React.Component {
    
}
## Basic usage

<Playground wrapper={Wrapper}>
    <Select
        placeholder="Choose your step"
        items={items}
    />
</Playground>

## Selected title prop usage

<Playground wrapper={Wrapper}>
    <Select
        selectedTitle="Drink"
        placeholder="Choose your step"
        items={items}
    />
</Playground>

## Search usage

<Playground wrapper={Wrapper}>
    <Select
        placeholder="Choose your step"
        items={items}
        isSearch={true}
    />
</Playground>
