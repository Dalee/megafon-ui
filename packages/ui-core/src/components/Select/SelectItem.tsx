import * as React from 'react';
import * as PropTypes from 'prop-types';
import './SelectItem.less';
import cnCreate from 'utils/cnCreate';

interface ISelectItemProps {
    /** Content */
    content?: JSX.Element[] | Element[] | JSX.Element | Element | string;
    /** Ordinal number of the item  */
    value: number;
    /** Current item */
    isCurrent?: boolean;
    /** Active item in dropdown list */
    isActive?: boolean;
    /** Item index */
    index?: number;
    /** Select item handler */
    onSelect?(e: React.SyntheticEvent<EventTarget>, value: number): void;
    /** Hover item handler */
    onHover?(e: React.SyntheticEvent<EventTarget>, value: number): void;
}

const cn = cnCreate('mfui-select-item');
class SelectItem extends React.Component<ISelectItemProps, {}> {
    static propTypes = {
        value: PropTypes.number.isRequired,
        content: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
            PropTypes.element,
        ]),
        isCurrent: PropTypes.bool,
        isActive: PropTypes.bool,
        index: PropTypes.number,
        onSelect: PropTypes.func.isRequired,
        onHover: PropTypes.func.isRequired,
    };

    itemNode: any = null;

    shouldComponentUpdate(nextProps: ISelectItemProps) {
        const { isActive, isCurrent, content } = this.props;
        const active = isActive !== nextProps.isActive;
        const current = isCurrent !== nextProps.isCurrent;
        const contentItem = content !== nextProps.content;

        return active || current || contentItem ;
    }

    handleClick = e => {
        const { onSelect, value } = this.props;

        onSelect && onSelect(e, value);
    }

    handleMouseEnter = e => {
        const { onHover, index } = this.props;

        index !== undefined && onHover && onHover(e, index);
    }

    render() {
        const { isCurrent, isActive, content } = this.props;

        return (
            <div
                className={cn('', {
                    current: isCurrent,
                    active: isActive,
                })}
                onClick={this.handleClick}
                onMouseEnter={this.handleMouseEnter}
                ref={node => { this.itemNode = node; }}
            >
                <div className={cn('title')}>
                    {content}
                </div>
            </div>
        );
    }
}

export default SelectItem;
