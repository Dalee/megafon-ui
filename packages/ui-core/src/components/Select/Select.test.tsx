import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Select from './Select';

const props = {
    className: 'test-class',
    label: 'test-label',
    labelId: '1',
    noticeText: 'test-notice-text',
    required: true,
    placeholder: 'test-placeholder',
    classes: {
        control: 'test-control-class',
        root: 'test-root-class',
    },
    items: [
        {
            value: 1,
            title: 'test-name-1',
        },
        {
            value: 2,
            title: 'test-name-2',
        },
    ],
};

const controlSelector = '.mfui-select__control';
describe('<Select />', () => {
    it('it renders Select', () => {
        const wrapper = mount(<Select items={props.items} />);
        expect(wrapper).toMatchSnapshot();
    });

    it.only('it renders with disabled state', () => {
        const handleClick = jest.fn();
        const wrapper = shallow(<Select items={[]} isDisabled />);

        wrapper.find(controlSelector).simulate('click');
        expect(handleClick).not.toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
    });
});
