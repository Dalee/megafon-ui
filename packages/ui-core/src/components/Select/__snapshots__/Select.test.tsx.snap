// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> it renders with disabled state 1`] = `
<div
  className="mfui-select mfui-select_disabled mfui-select_no-touch"
>
  <div
    className="mfui-select__inner"
  >
    <div
      className="mfui-select__control"
      onKeyDown={[Function]}
    >
      <div
        className="mfui-select__title"
        onClick={[Function]}
        tabIndex={0}
      >
        <div
          className="mfui-select__title-inner"
        />
      </div>
      <div
        className="mfui-select__arrow-wrap"
        onClick={[Function]}
        tabIndex={1}
      >
        <span
          className="mfui-select__arrow"
        />
      </div>
    </div>
    <div
      className="mfui-select__list"
    >
      <div
        className="mfui-select__list-inner"
      >
        <div
          className="mfui-select__not-found"
        >
          Ничего не нашлось
        </div>
      </div>
    </div>
  </div>
</div>
`;
